### DISCUSSION TITLE
Logging for Docker Swarm?

### DISCUSSION THREAD

**Discussion by 1c7** [BretFisher/ama#161] [Q&A]
Hi Bret~
I am about to use Docker Swarm in production. 
for now, I am just using a demo app to test all sort of things about Docker Swarm to get familiar with it.

## Question: What logging option do you recommend with Docker Swarm?
I did some research before I post this issue.

seem like the only option is:
* [docker-elk](https://github.com/deviantony/docker-elk)

but it's a bit complex.
I don't know how to config my Ruby on Rails app to send log to Logstash.
and How to use Kibana.

Have you consider make a couple video about logging 
And put them on Youtube or your Udemy swarm course?

## FYI, using a 1-node Swarm. 8G memory.

Thanks!

**Comment by 1c7** [comment]
by the way Thanks for both course on Udemy. 
docker mastery and swarm mastery. 
Really helpful stuff
also I watched a lot of your Youtube video.

**Comment by BretFisher** [comment]
Logging videos are planned for Swarm Mastery but they will be mostly ELK as that's the top open source solution right now. If you don't want to host your own logging, you can use my favorite SaaS [PaperTrail (this link gets you a free extra GB's of storage)](https://papertrailapp.com/?thank=e726ed). Remember that docker logging means you're using a logging driver when creating your Swarm Services (or just docker run) to point the output of your "docker service logs" commands to the proper logging service.
You can see in my swarm samples repo, I set default logging for all services [here in a template block](https://papertrailapp.com/?thank=e726ed), and then inject that config on line 15. You could change that config to point to logstash port, or to papertrail.

**Comment by 1c7** [comment]
Thank you
